src/components/KTreeList/KTreeDraggable.vue

```diff
@@ -0,0 +1,345 @@
+ *  Note: if TS errors appear on the
+ *  <template #[itemIcon]="slotProps"> or
+ *  <template #[itemLabel]="slotProps">
+ * lines switch to `v-slot:` notation and Save to let linter clear -->
+ */
+import { computed, ref, watch, onMounted, PropType } from 'vue'
+import { VueDraggableNext } from 'vue-draggable-next'
+import KTreeItem, { TreeListItem } from '@/components/KTreeList/KTreeItem.vue'
+
+export interface ChangeEvent {
+  items: TreeListItem[]
+}
+
+export interface ChildChangeEvent {
+  parent: string,
+  children: TreeListItem[]
+}
+
+export interface DropEvent {
+  item: TreeListItem,
+  parent: TreeListItem
+}
+
+const props = defineProps({
+  items: {
+    type: Array as PropType<TreeListItem[]>,
+    required: true,
+    validator: (items: TreeListItem[]) => !items.length || items.every(i => i.name !== undefined && i.id !== undefined),
+  },
+  disableDrag: {
+    type: Boolean,
+    default: false,
+  },
+  maxLevel: {
+    type: Number,
+    default: 2,
+  },
+  /**
+   * These are internal props used to track info
+   * of nested children.
+   */
+  level: {
+    type: Number,
+    default: 0,
+  },
+  parentId: {
+    type: String,
+    default: undefined,
+  },
+})
+
+const emit = defineEmits<{
+  (event: 'change', data: ChangeEvent): void,
+  (event: 'child-change', data: ChildChangeEvent): void,
+  (event: 'selected', item: TreeListItem): void
+}>()
+
+const key = ref(0)
+const internalList = ref<TreeListItem[]>([])
+const draggableAttrs = {
+  tag: 'div',
+  // Note: force-fallback: true is required to override cursor when dragging
+  'force-fallback': true,
+  animation: '100',
+  draggable: '.k-tree-item-container',
+  'item-key': 'id',
+  'ghost-class': 'k-tree-item-dragged',
+  'drag-class': 'k-tree-item-grabbing',
+  class: 'child-drop-zone',
+}
+const dragging = ref(false)
+// using these vars to trick the TS compiler and avoid errors about circular refs in the <template>
+const itemIcon = 'item-icon'
+const itemLabel = 'item-label'
+
+const iconSecondaryColor = (item: TreeListItem): string | undefined => {
+  if (item.icon === 'treeDoc' || !item.icon) {
+    return item.selected ? 'var(--KTreeListItemSelectedBorder, var(--teal-200))' : 'var(--KTreeListItemUnselectedBorder, var(--grey-200))'
+  }
+
+  return undefined
+}
+
+const hasNoChildren = (item: TreeListItem): boolean => {
+  return !internalList.value.filter(anItem => anItem.id === item.id)?.[0].children?.length
+}
+
+const getElementChildren = (item: TreeListItem): TreeListItem[] => {
+  return item.children as TreeListItem[]
+}
+
+const handleChangeEvent = (): void => {
+  if (props.parentId) {
+    emit('child-change', {
+      parent: props.parentId,
+      children: internalList.value,
+    })
+  } else {
+    emit('change', {
+      items: internalList.value,
+    })
+  }
+}
+
+const handleSelectionEvent = (item: TreeListItem) => {
+  emit('selected', item)
+}
+
+const maxLevelReached = computed((): boolean => {
+  return props.level > props.maxLevel
```

Comment:
I'm not exactly sure, but is this correct or should it be `>=` ?

Reply:
It works with the current implementation so I'm not going to change this.